ods_ci/tests/Tests/0200__rhoai_upgrade/0203__post_upgrade.robot

```diff
@@ -212,6 +219,47 @@ Verify that the must-gather image provides RHODS logs and info
     [Teardown]  Cleanup must-gather Logs
 
 
+Verify That DSC And DSCI Release.Name Attribute matches required value
+    [Documentation]    Tests the release.name attribute from the DSC and DSCI matches the desired value.
+    ...                ODH: Open Data Hub
+    ...                RHOAI managed: OpenShift AI Cloud Service
+    ...                RHOAI selfmanaged: OpenShift AI Self-Managed
+    [Tags]    Upgrade
+    ${DSC_RELEASE_NAME}=     Get Resource Attribute      ${OPERATOR_NAMESPACE}
+    ...                      DataScienceCluster       ${DSC_NAME}        ${RELEASE_NAME_ATTRIBUTE_PATH}
+    ${DSCI_RELEASE_NAME}=     Get Resource Attribute      ${OPERATOR_NAMESPACE}
+    ...                      DSCInitialization      ${DSCI_NAME}        ${RELEASE_NAME_ATTRIBUTE_PATH}
+
+    IF    "${PRODUCT}" == "RHODS"
+        Run Keyword If RHODS Is Self-Managed      Should Be Equal As Strings    ${DSC_RELEASE_NAME}     ${RHOAI_SELFMANAGED_RELEASE_NAME}
+        Run Keyword If RHODS Is Self-Managed      Should Be Equal As Strings    ${DSCI_RELEASE_NAME}    ${RHOAI_SELFMANAGED_RELEASE_NAME}
+        Run Keyword If RHODS Is Managed        Should Be Equal As Strings    ${DSC_RELEASE_NAME}     ${RHOAI_MANAGED_RELEASE_NAME}
+        Run Keyword If RHODS Is Managed        Should Be Equal As Strings    ${DSCI_RELEASE_NAME}    ${RHOAI_MANAGED_RELEASE_NAME}
+    ELSE IF    "${PRODUCT}" == "ODH"
+        Should Be Equal As Strings    ${DSC_RELEASE_NAME}     ${ODH_RELEASE_NAME}
+        Should Be Equal As Strings    ${DSCI_RELEASE_NAME}    ${ODH_RELEASE_NAME}
+    END
```

Comment:
I think you could make it a bit more readable by moving the IF/ELSE statement in Suite Setup.
In order to check the resource attribute, you'd then do some like `DSC Release Name Should be ${expected_name}`

Reply:
will give a try

Reply:
added